### DISCUSSION TITLE
Integration only returns 2000 features

### DISCUSSION THREAD

**Discussion by YunnnQin** [satijalab/seurat#7405] [General]
Hi everyone! I'm using Seurat to integrate my data. I have two datasets, one from human, the other one from human. I have already make sure they have the same genes, which 14736 features.  
The last time I try the integration works well for me. It will return the same number of genes in the integrated data. But this time, I run the same code, but it only returns 2000 genes. And also I have tried the code on another dataset, every time it only keeps 2000 genes. Is anyone have the same issue? Or is there something wrong I did? Thanks for your time!

Here are my code:
# Put object in a list
Int_hp_list <- list()
Int_hp_list[['human']] <- human.integrated.common
Int_hp_list[['pig']] <- pig.adult.common

# Normalize and find variable genes in each samples
Int_hp_list <- lapply(X=Int_hp_list, FUN=function(x){
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x,selection.method='vst', nfeatures =2000)
})

# select features that are repeatedly variable across datasets for integration 
features <- SelectIntegrationFeatures(object.list = Int_hp_list)

# find anchor and integrate
hp.anchors <- FindIntegrationAnchors(object.list = Int_hp_list,anchor.features = features)

# create integrated data assay
hp.integrated <- IntegrateData(anchorset = hp.anchors)

<img width="649" alt="Screenshot 2023-06-02 at 14 18 16" src="https://github.com/satijalab/seurat/assets/93378251/cc7ab413-c2f5-48e8-9eec-7b6299fd0dcd">

Best,
Yun

**Comment by mattoonerin** [comment]
Hi @YunnnQin,

I'm experiencing the same issue with my dataset. Did you have any updates on how your resolved this?

Thank you!
Erin

**Comment by lizchcase** [comment]
The default for SelectIntegrationFeatures is nfeatures=2000. This is because it works on just the most variable features since this should recapitulate most of the variance in the data. You can change that parameter if you want to include more features, but know that it will take more time/computational power. If you want to include all the genes present in both objects, you can set nfeatures = length(intersect(rownames(object1),rownames(object2))

**Comment by sam-israel** [comment]
@YunnnQin 

> The last time I try the integration works well for me. It will return the same number of genes in the integrated data. But this time, I run the same code, but it only returns 2000 genes.

Maybe you are confusing the Integrated and the SCT assays. 

`nrow(intergrated_seurat_object@assays$integrated)`

should return only the 2000 number of genes which was specified as number of variable features.
But 

`nrow(intergrated_seurat_object@assays$SCT)` 

should be all genes imho.
